src/js/grapl-cdk/lib/engagement.ts

```diff
@@ -0,0 +1,253 @@
+                endpointExportName: "EngagementEndpointApi",
+            },
+        );
+    }
+}
+
+export class EngagementNotebook extends cdk.NestedStack {
+    securityGroup: ec2.SecurityGroup;
+    connections: ec2.Connections;
+
+    constructor(
+        scope: cdk.Construct,
+        id: string,
+        prefix: string,
+        user_auth_db: UserAuthDb,
+        vpc: ec2.Vpc,
+    ) {
+        super(scope, id);
+
+        this.securityGroup = new ec2.SecurityGroup(
+            this,
+            prefix + '-notebook-security-group',
+            { vpc: vpc });
+
+        this.connections = new ec2.Connections({
+            securityGroups: [this.securityGroup],
+            defaultPort: ec2.Port.allTcp()
+        });
+
+        const role = new iam.Role(
+            this,
+            id + 'notebook-role',
+            {
+                assumedBy: new iam.ServicePrincipal('sagemaker.amazonaws.com')
+            }
+        );
+
+        user_auth_db.allowReadWriteFromRole(role);
+
+        const _notebook = new sagemaker.CfnNotebookInstance(
+            this,
+            id + '-sagemaker-endpoint',
+            {
+                instanceType: 'ml.t2.medium',
+                securityGroupIds: [this.securityGroup.securityGroupId],
+                subnetId: vpc.privateSubnets[0].subnetId,
+                directInternetAccess: 'Enabled',
+                roleArn: role.roleArn
+            }
+        );
+
+    }
+}
+
+export class EngagementUx extends cdk.Stack {
+    constructor(
+        scope: cdk.Construct,
+        id: string,
+        prefix: string,
+        edge: EngagementEdge,
+        graphql_endpoint: GraphQLEndpoint,
+    ) {
+        super(scope, id + 'Stack', { stackName: 'Grapl-EngagementUX' });
+
+        const bucketName = `${prefix}-engagement-ux-bucket`;
+
+        const edgeBucket = new s3.Bucket(this, bucketName, {
+            bucketName,
+            publicReadAccess: true,
+            websiteIndexDocument: 'index.html',
+            removalPolicy: RemovalPolicy.DESTROY,
+        });
+
+        getEdgeGatewayId(
+            [edge.integrationName, graphql_endpoint.integrationName],
+            (loginGatewayId: string, graphQLGatewayId: string) => {
+                const srcDir = path.join(__dirname, "../edge_ux/");
+                const packageDir = path.join(__dirname, "../edge_ux_package/");
+
+                if (!fs.existsSync(packageDir)) {
+                    fs.mkdirSync(packageDir);
+                }
+
+                const loginUrl = `https://${loginGatewayId}.execute-api.${aws.config.region}.amazonaws.com/prod/`;
+                const graphQLUrl = `https://${graphQLGatewayId}.execute-api.${aws.config.region}.amazonaws.com/prod/`;
+
+                const replaceMap = new Map();
+                replaceMap.set(`http://"+window.location.hostname+":8900/`, loginUrl);
+                replaceMap.set(`window.location.host+":5000/"`, graphQLUrl);
+
```

Comment:
This doesn't require the http://+? Is that because it's already in the code?

Reply:
Hm. Looks like this is correct, as far as I can tell (not well), though not sure as this was merged from staging. I imagine it's intended to replace the following:

`h=function(){return window.location.host+":5000/"},g=window.location.host+":8900/"`

That snippet suggests there should be similar replace map like `replaceMap.set(`window.location.host+":8900/"`, loginUrl);`

I see what looks like there should also be a `replaceMap.set(`http://"+window.location.hostname+":5000/`, graphQLUrl);`

so I'll add those. Maybe @andrea-grapl can verify.

Reply:
`"http://"+window.location.hostname+":5000/"`

Reply:
Nevermind, per our off-line discussion and rebuild of the edge_ux, I no longer see that first snippet I included in my last comment.